api/src/http/routes/change-password.ts

```diff
@@ -23,8 +26,51 @@ export async function changePassword(server: FastifyInstance) {
             id: number;
         };
 
-        const profile = await profileRepository.findById(decoded.id);
+        let changePassword: ChangePasswordInput;
 
-        return reply.send(profile);
+        try {
+            changePassword = changePasswordSchema.parse(request.body);
+        } catch (error) {
+            return reply.status(400).send({
+                message: (error as Error).message,
+            });
+        }
+
+        const { currentPassword, newPassword, repeatPassword } = changePassword;
+
+        const profile = await profileRepository.findById(decoded.id, true);
+
+        if (!profile) {
+            return reply
+                .status(404)
+                .send({ message: "Usuário não encontrado" });
+        }
+
+        const passwordMatch = await bcrypt.compare(
+            currentPassword,
+            profile.password,
+        );
```

Comment:
Só mudando o local de onde faz a descontrução pra ficar mais próximo de onde começa a ser usado, assim não fica parecendo que tá jogado.

```suggestion
        const profile = await profileRepository.findById(decoded.id, true);

        if (!profile) {
            return reply
                .status(404)
                .send({ message: "Usuário não encontrado" });
        }
        
         const { currentPassword, newPassword, repeatPassword } = changePassword;

        const passwordMatch = await bcrypt.compare(
            currentPassword,
            profile.password,
        );
```